src/Microsoft.Health.Dicom.Blob/Extensions/AzureStorageErrorExtensions.cs

```diff
@@ -0,0 +1,35 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Azure;
+
+namespace Microsoft.Health.Dicom.Blob.Extensions;
+
+internal static class AzureStorageErrorExtensions
+{
+    private static readonly List<string> Customer403ErrorCodes = new List<string>
+    {
+        "AuthorizationPermissionMismatch",
+        "AuthorizationFailure",
+        "KeyVaultEncryptionKeyNotFound",
+    };
+
+    private const string FileSystemNotFoundErrorCode = "FilesystemNotFound";
+
+    public static bool IsConnectedStoreCustomerError(this RequestFailedException rfe)
+    {
+        return (rfe.Status == 403 && Customer403ErrorCodes.Contains(rfe.ErrorCode)) ||
+            (rfe.Status == 404 && rfe.ErrorCode.Equals(FileSystemNotFoundErrorCode, StringComparison.OrdinalIgnoreCase)));
+    }
+
+    public static bool IsStorageAccountUnknownError(this Exception exception)
```

Comment:
FilestorageNotFound happens when you use the data lake client to interact with the storage account & the container is not found. I tested with both the data lake client & the blob client. Blob will use "ContainerNotFound" for this. Added both as possible error codes

Reply:
The error depends on which client you use today, so despite it being HNS enabled, BlobClient will always throw "ContainerNotFound" exception.

If you use `DataLakeServiceClient`, which is only for HNS enabled storage accounts, it will throw "FilestorageNotFound" exception. Today we should never see that, but I wanted to leave it in just in case something changed or if we ever used a different client, since it is specific to HNS accounts